### ISSUE TITLE
Linter.py: Also check for REQUIREMENTS

### ISSUE THREAD

**Issue by Makman2** [coala/coala#3983]
`@linter` currently only checks for the executable and ignores the check of the base `Bear`. This should be implemented.

One thing:
I'm not sure whether the old check for the executable should be removed. @sils stated this in one of my outdated PRs:

>  sils on Oct 7, 2016 Member
>
> do we need this check at all? The requirements check should be totally sufficient
> @Makman2

>Makman2 on Oct 8, 2016 Member
>
>already thought about this too. Will do that, though that interferes with #2073 a bit^^

Although I said "Will do that", I think it's a good safety check we should maybe keep.

Another thing:
The best is to refer to the base class' `check_prerequisites`. This may get a bit complicated as the linter-type is constructed at run-time, and also the inheritance is dynamic (either `LocalBear` or `GlobalBear` depending what the user wants). I hope using `super()` works fine.

**Comment by Siddhartha1234** [comment]
@Makman2 super should work fine. Should I add it as an error message ?

**Comment by Makman2** [comment]
> Should I add it as an error message ?

What do you mean?

**Comment by jayvdb** [comment]
@Makman2, @Siddhartha1234, @srisankethu, @arush0311 , it seems this is a dup of https://github.com/coala/coala/pull/3526 , or at least tightly related... ?

**Comment by arush0311** [comment]
@jayvdb yes I also did the changes done in this PR but there were some other changes also

**Comment by Makman2** [comment]
Hm yeah looks like, could you @arush0311 and @Siddhartha1234 sync up with each other? :) Then we decide what we are going to close :+1:

**Comment by arush0311** [comment]
I think we can merge this first. https://github.com/coala/coala/pull/3526 will need more work :D

**Comment by jayvdb** [comment]
Note if this doesnt get merged into release 0.11 , there will be a [better dependency tracking tool](https://gitlab.com/coala/package_manager/issues/75) coming from upstream.

**Comment by Makman2** [comment]
@sils if this issue is not fixed, please reopen.

**Comment by sils** [comment]
Yep forgot reopening. See revert and the fix for the stuff.

On Jul 15, 2017 18:58, "Mischa Krüger" <notifications@github.com> wrote:

@sils <https://github.com/sils> if this issue is not fixed, please reopen.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/coala/coala/issues/3983#issuecomment-315547312>, or mute
the thread
